pkg/certificate/manager.go

```diff
@@ -111,23 +114,34 @@ func (m *Manager) EnsureCertificate(certReq Request, ownerName string) (Certific
 	gid, _ := util.GetGID(constant.Group)
 	uid, _ := util.GetUID(ownerName)
 
-	if util.FileExists(keyFile) && util.FileExists(certFile) {
-		_ = os.Chown(keyFile, uid, gid)
-		_ = os.Chown(certFile, uid, gid)
+	var certsExist bool
 
-		cert, err := ioutil.ReadFile(certFile)
-		if err != nil {
-			return Certificate{}, errors.Wrapf(err, "failed to read ca cert %s for %s", certFile, certReq.Name)
-		}
-		key, err := ioutil.ReadFile(keyFile)
-		if err != nil {
-			return Certificate{}, errors.Wrapf(err, "failed to read ca key %s for %s", keyFile, certReq.Name)
-		}
+	// this loop lets regenerateCert asses if a certificate needs to be generated
+	// certsExist is a dummy value for a single-run loop
+
+	for certsExist = true; certsExist; {
```

Comment:
Couldn't we achieve the same logic without the loop around if we do something like:
```
if !m.regenerateCert(certReq, certFile) {
   // Load and return here, the same stuff that happens in the else block now
}

```